### PR TITLE
isolate internal function

### PR DESCRIPTION
**Pull request by j8xixo12**
I  found `devenv` internal function can be accessed by user, it will import internal function to user’s session.
I modified each cmd and some helper function to avoid this.

### CODE REVIEW

**Comment by yungyuc** [comment]
Sounds like a good idea.  Could you annotate the diff to make it easier for me to understand where the names leak?

**Review by tai271828 — APPROVED** [review]
Looks like the trick is to use `get_list` to retrieve sub-commands which are renamed with underscore prefix. The trick looks good to me (no downside comes to my mind). I would +1 to merge but will be also glad to see more annotation for me to dive into more details easily.

**Comment by j8xixo12** [comment]
@yungyuc @tai271828  In the beginning, I thought name leak happened in `cmd.d`, I was wrong, the cmds in `cmd.d`  are called with **new bash session** [here](https://github.com/solvcon/devenv/blob/main/bin/devenv#L16), so name leak should not happen!

Name leaks happened in `scripts/init`, because user need to source `scripts/init`, the function [devenv](https://github.com/solvcon/devenv/blob/main/scripts/init#L5), [_devenv](https://github.com/solvcon/devenv/blob/main/scripts/init#L48) will source `scripts/func.d/bash_utils` and `scripts/env.d/devenv_impl`, name leak happens at this time.

So I use subshell technique to avoid `scripts/env.d/devenv_impl` name leak, but `namemunge`, `nameremove` these two functions need to manipulate env variables in user session, therefore subshell technique will not work, I unset all functions in `scripts/env.d/devenv_impl`.

**Review by tai271828 — APPROVED** [review]

File: scripts/init [hunk 1]
```diff
@@ -3,50 +3,60 @@ export DEVENVROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
...
-  else
-    ${DEVENVROOT}/bin/devenv "$@"
-  fi
+      export DEVENVCURRENTROOT="/dev/null"
+      ;;
+    *)
+      ${DEVENVROOT}/bin/devenv "$@"
+      ;;
+  esac
+  unset -f nameremove
```
Comment (yungyuc) [inline scripts/init:54]: All devenv functions should be prefixed with something, e.g., `devenv_` or `_devenv_`.  Or maybe we would unset names already defined by the users.

`nameremove` and `namemunge` are two of the utility functions I always add in my own bashrc.  Please do not unset them.

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: scripts/init (hunk 1 above)
Comment (j8xixo12) [reply scripts/init]: I just added `devenv` as prefix in `devenv_impl`.

Comment (yungyuc) [reply scripts/init]: Thanks.  I guess you marked it resolved after pushing.  Next time please leave it there so that I can keep track of the point.

**Review by yungyuc — APPROVED** [review]

**Comment by yungyuc** [comment]
LGTM

**Comment by chestercheng** [comment]
LGTM
i was amazed by this trick, i never thought subshell could be used like this.

**Review by chestercheng — APPROVED** [review]

**Pull request merged**